rust/lance/src/index/append.rs

```diff
@@ -15,56 +15,73 @@
 use std::sync::Arc;
 
 use lance_core::{format::Index as IndexMetadata, Error, Result};
-use lance_index::scalar::lance_format::LanceIndexStore;
-use lance_index::IndexType;
-use log::info;
+use lance_index::{optimize::OptimizeOptions, scalar::lance_format::LanceIndexStore, IndexType};
 use roaring::RoaringBitmap;
 use snafu::{location, Location};
 use uuid::Uuid;
 
+use super::vector::ivf::optimize_vector_indices;
+use super::DatasetIndexInternalExt;
 use crate::dataset::scanner::ColumnOrdering;
 use crate::dataset::Dataset;
-use crate::index::vector::ivf::IVFIndex;
-
-use super::DatasetIndexInternalExt;
 
 /// Append new data to the index, without re-train.
 ///
 /// Returns the UUID of the new index along with a vector of newly indexed fragment ids
```

Comment:
Let's update this comment and explain how it works.  It will not be clear to a reader how an "append" operation distributes the new data amongst the existing indices.  In fact, is this still an append?  Or is it an "append + merge?"

I am a little confused by this function.

Reply:
I updated the doc; lemme me know what you think.  @westonpace